tempesta_fw/http_parser.c

```diff
@@ -887,8 +1081,38 @@ __FSM_STATE(RGen_OWS) {							\
 	goto fsm_reenter;						\
 }
 
+/*
+ * Save parsed data to list of raw hop-by-hop headers if data doesn't match
+ * to @name and do @lambda otherwize
+*/
+#define TRY_HBH_TOKEN(name, lambda)					\
+	TRY_STR_LAMBDA_finish(name, lambda, {				\
+		if (__hbh_parser_add_data(hm, data, len, false))	\
+			r = CSTR_NEQ;					\
+	}, I_EoT)
+
+/* Add spec header of id @hid and @name to list of spec hop-by-hop headers */
+#define TRY_HBH_SPEC_HDR_LAMBDA(name, hid, lambda)			\
+	TRY_HBH_TOKEN(name, {						\
+		BUILD_BUG_ON(sizeof(parser->hbh_parser.spec) * CHAR_BIT	\
+			     <  TFW_HTTP_HDR_RAW);			\
+		lambda;							\
+		parser->hbh_parser.spec |= 0x1 << hid;			\
+		if (!TFW_STR_EMPTY(&msg->h_tbl->tbl[hid]))		\
+			msg->h_tbl->tbl[hid].flags |= TFW_STR_HBH_HDR;	\
+	})
```

Comment:
The macro is used in only one place, so the code would be more readable if you place the macro as inline code at the appropriate place.